src/EnergyPlus/VariableSpeedCoils.cc

```diff
@@ -5829,6 +5834,49 @@ namespace VariableSpeedCoils {
                 ShowContinueError(state, "... to ensure they meet the expected manufacturers performance specifications.");
             }
         }
+
+        Array1D<DataHeatBalance::RefrigCondenserType> CondenserType;
+        StandardRatings::HPdefrostControl DefrostControl;
+        switch (varSpeedCoil.VSCoilType) {
+        case DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed:
+            CondenserType.push_back(varSpeedCoil.CondenserType);
+            switch (varSpeedCoil.DefrostControl) // defrost control; 1=timed, 2=on-demand
+            {
+            case 1:
+                DefrostControl = StandardRatings::HPdefrostControl::Timed;
+                break;
+            case 2:
+                DefrostControl = StandardRatings::HPdefrostControl::OnDemand;
+                break;
+            default:
+                break;
+            }
+            StandardRatings::CalcDXCoilStandardRating(state,
```

Comment:
The magic numbers in here are ... not great, but I don't think you are to blame for them.  That aside, is there a chance the nested switch block above could go through the default section?  If so, what would DefrostControl be?  Assuming timed is the default, what about 

```c++
StandardRatings::HPdefrostControl DefrostControl = StandardRatings::HPdefrostControl::Timed;
if (varSpeedCoil.DefrostControl == 2) {
  DefrostControl = StandardRatings::HPdefrostControl::OnDemand;
}
```

Reply:
I've updated the switch block to have DefrostControl as Timed in case of default :
![image](https://github.com/NREL/EnergyPlus/assets/78803858/7f46d5cd-42a6-4a15-9137-fc849831bdac)